packages/block-library/src/navigation/index.php

```diff
@@ -135,7 +135,16 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 	);
 
 	$inner_blocks_html = '';
+	$is_list_open = false;
 	foreach ( $block->inner_blocks as $inner_block ) {
+		if ($inner_block->name === "core/navigation-link" && $is_list_open === false) {
+			$is_list_open = true;
+			$inner_blocks_html .= '<ul class="wp-block-navigation__container">';
+		}
+		if ($inner_block->name !== "core/navigation-link" && $inner_block->name !== "core/spacer" && $is_list_open === true) {
+			$is_list_open = false;
+			$inner_blocks_html .= '</ul>';
+		}
```

Comment:
Nice, much closer to expected html markup. If we go with this approach, children of ul, still need to be wrapped in an li. (For example the spacer div).

<img width="1080" alt="Screen Shot 2021-04-26 at 8 59 14 AM" src="https://user-images.githubusercontent.com/1270189/116113852-bdbf2980-a66d-11eb-9a65-debb5c9b8559.png">